archive/compression/compression_test.go

```diff
@@ -189,3 +189,31 @@ func TestCmdStreamBad(t *testing.T) {
 		t.Fatalf("wrong output: %s", string(buf))
 	}
 }
+
+func TestDetectCompressionZstd(t *testing.T) {
+	// test zstd compression without skippable frames.
+	compressedData := []byte{
```

Comment:
For these sort of test cases, best to just set them up beforehand. Since the skippable has the range then also test the boundary conditions.

I often set them up like this...
```.go
for _, tc := range []struct{
	source []byte
	expected Compression
} {
	{
		source: []byte{0x28, 0xb5, 0x2f, 0xfd, .....},
		expected: Zstd,
	},
} {
	compression := DetectCompression(tc.source)
	if compression != tc.expected {
		t.Fatal("Unexpected compression %s, expected %s", compression, tc.expected)
	}
}
```

Reply:
Done.